File: create_redundancy.py
#!/usr/bin/python
# Filename : create_redundancy.py
# Create redundancy script
# Coded by Steve Moss - gawbul@gmail.com

# Define modules to import
import sys # could use from sys import argv here

# Check number of command-line arguments are correct
if len(sys.argv) != 4:
	# Incorrect usage
	print 'Usage: ./create_redundancy.py <infile> <redundancy> <outfile>\n'
else:
	# Load infile and output to outfile redundancy number of times
	infile = open(sys.argv[1], 'r')
	buffer = infile.read()
	infile.close()
	
	# Begin redundancy loop
	for i in range(1, int(sys.argv[2]) + 1):
		outfile = open(sys.argv[3], 'a')
		outfile.write(buffer)
	
	# Finish up, close files
	outfile.close()
	print 'Done'

File: create_redundancy.pl
#!/usr/bin/perl
#------------------------------------------------------------------------#
# Perl script to create multiple redundancy in a fasta file for testing  #
# external redundancy scripts.                                           #
# Coded by Steve Moss                                                    #
# Email: gawbul@gmail.com					                             #
#------------------------------------------------------------------------#

use strict;
use warnings;

# define variables
my $filename="";
my $redundancy="";
my $output="";
my $buffer="";

# Check for valid usage; if incorrect number of arguments then display usage and exit
if($#ARGV != 2) {
	print("Usage: ./create_redundancy.pl <infile> <redundancy> <outfile>\n");
}
else {
	# Assign variables
	$filename = $ARGV[0];
	$redundancy = $ARGV[1];
	$output = $ARGV[2];
	
	# Open file and chomp to ensure things are read in properly
	# Read file to buffer
	undef $/;
	open(FAS_IN, $filename) or die "Error: Couldn't open $filename!";
	$buffer = <FAS_IN>;
	close(FAS_IN);
	
	# Create redundancy in file using number given in command-line arguments
	for (my $i = 1; $i <= $redundancy; $i++) {
		open(FAS_OUT, ">>$output");
		print FAS_OUT $buffer;
	}
	close FAS_OUT;
	print "Done\n";
}
